### PR TITLE
Mh user profile

### PR DESCRIPTION
**Pull request by merrillhuang**
Merge user profile branch into main.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -22,10 +22,11 @@ def edit
   # POST /comments or /comments.json
   def create
     @comment = Comment.new(comment_params)
+    @comment.author = current_user
 
     respond_to do |format|
       if @comment.save
-        format.html { redirect_to comment_url(@comment), notice: "Comment was successfully created." }
+        format.html { redirect_back fallback_location: root_path, notice: "Comment was successfully created." }
```
Comment (marywebby) [inline app/controllers/comments_controller.rb:29]: good job changing it to redirect_back !

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by!(username: params.fetch("username"))
```
Comment (marywebby) [inline app/controllers/users_controller.rb:3]: Be wary of when to be using `"username"` and when to be using `:username`

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                <div class="flex-grow-1 ms-3">
+                  <h5 class="mt-0"><%= comment.author.username %></h5>
+                  <p><%= comment.body %></p>
+                </div>
+              </div>
+            </li>
+          <% end %>
+        </ul>
+        <div class="card-body">
+          <%= render partial: "comments/form", locals: { comment: Comment.new(photo_id: photo.id) } %>
```
Comment (marywebby) [inline app/views/users/show.html.erb:30]: This is good! There is another way of doing this such as `<%= render "comments/form", comment: Comment.new(photo_id: photo.id) %>`. The way you did it is correct, just makes it more explicit, but if you're looking for a shorter version the one i provided also works.

**Review by marywebby — APPROVED** [review]
Nice job Merrill! Just make sure to get a code review from a peer!

File: config/routes.rb [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
   resources :likes
   resources :follow_requests
   resources :comments
   resources :photos
-  devise_for :users
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
-  # Defines the root path route ("/")
-  root "photos#index"
+  get "/:username" => "users#show", as: :user
```
Comment (jessjno) [inline config/routes.rb:13]: looks good

**Review by jessjno — COMMENTED** [review]
looks good!

File: app/views/comments/_form.html.erb [hunk 5]
```diff
@@ -12,21 +12,14 @@
...
-    <%= form.label :body, style: "display: block" %>
-    <%= form.text_area :body %>
+  <div class="mb-3">
+    <%= form.text_area :body, class: "form-control" %>
   </div>
 
   <div>
-    <%= form.submit %>
+    <%= form.submit class: "btn btn-outline-secondary btn-block" %>
   </div>
```
Comment (Ritumka) [inline app/views/comments/_form.html.erb:24]: It's great that you can automatically generate a template of the form and just change it as you need!

File: config/routes.rb [hunk 6]
```diff
@@ -1,11 +1,14 @@
 Rails.application.routes.draw do
+  # Defines the root path route ("/")
+  root "photos#index"
```
Comment (Ritumka) [inline config/routes.rb:3]: I also think it's better to write routes in the order you actually open pages in your app. Great job!

**Review by Ritumka — COMMENTED** [review]
Great job!

**Pull request merged**